Zend/tests/fibers/failing-fiber.phpt

```diff
@@ -0,0 +1,23 @@
+--TEST--
+Test throwing into fiber
+--FILE--
+<?php
+
+$fiber = new Fiber(function (): void {
+    Fiber::suspend('test');
+    throw new Exception('test');
+});
+
+$value = $fiber->start();
+var_dump($value);
+
+$fiber->resume($value);
+
+--EXPECTF--
+string(4) "test"
+
+Fatal error: Uncaught Exception: test in %sfailing-fiber.php:%d
+Stack trace:
+#0 [internal function]: {closure}()
+#1 {main}
```

Comment:
Is there a way we could change `[internal function]` to something like `[fiber]` instead?

Reply:
One suggestion I have is swapping `{main}` to `{fiber}` in the exception backtrace: https://github.com/trowski/php-src/commit/82653197553e00699bb645d5b768a65ff44913f4

I attempted to save a pointer the call stack invoking start, but a fiber is able to long outlast the stack that may have started it. It's also undesirable to swap the stack trace on each resume, as this will drop the backtrace context in the fiber, which leads to the terrible backtraces we often had with Generators in Amp.

Reply:
> One suggestion I have is swapping {main} to {fiber} in the exception backtrace: trowski@8265319

Adding a bool for this seems really ad-hoc ... would it be valuable for something if the exception contains a reference to the containing Fiber object outright?

> I attempted to save a pointer the call stack invoking start, but a fiber is able to long outlast the stack that may have started it. It's also undesirable to swap the stack trace on each resume, as this will drop the backtrace context in the fiber, which leads to the terrible backtraces we often had with Generators in Amp.

Not sure I follow. To be clear, what I would expect is the backtrace at the point of resume, not at point of start. For generators, this is implemented with a dummy frame that gets resolved when the backtrace is actually fetched. Why are the resulting backtraces "terrible"?

Reply:
@nikic Including `resume()` doesn't really make sense, because you want the complete fiber stack there, where `resume()` would be somewhere in between. But having `{main}` there is really weird, I guess it should be `{fiber}`.